internal/pkg/gateway/registry.go

```diff
@@ -49,44 +50,38 @@ type endorserState struct {
 }
 
 // Returns a set of endorsers that satisfies the endorsement plan for the given chaincode on a channel.
-func (reg *registry) endorsers(channel string, chaincode string) ([]*endorser, error) {
-	err := reg.registerChannel(channel)
-	if err != nil {
-		return nil, err
-	}
-
+func (reg *registry) endorsers(channel string, interest *peer.ChaincodeInterest, preferOrg string) ([]*endorser, error) {
 	var endorsers []*endorser
+	var reserveEndorsers []*endorser
 
-	interest := &peer.ChaincodeInterest{
-		Chaincodes: []*peer.ChaincodeCall{{
-			Name: chaincode,
-		}},
-	}
-
-	descriptor, err := reg.discovery.PeersForEndorsement(common.ChannelID(channel), interest)
+	descriptor, err := reg.discovery.PeersForEndorsement(gossipcommon.ChannelID(channel), interest)
 	if err != nil {
-		return nil, err
+		logger.Errorf("PeersForEndorsement failed with error %s for channel %s and ChaincodeInterest %s", err, channel, proto.MarshalTextString(interest))
```

Comment:
```suggestion
		logger.Errorw("PeersForEndorsement failed.", "error", err, "channel", channel, "ChaincodeInterest", proto.MarshalTextString(interest))
```
For quite some time, we have been following this logging API for dumping logs in a better machine readable way.